packages/web-app-files/tests/unit/views/PublicLink.spec.js

```diff
@@ -1,34 +1,194 @@
+import PublicLink from '@files/src/views/PublicLink.vue'
+import { getStore, localVue } from './views.setup.js'
+import { shallowMount, mount } from '@vue/test-utils'
+
+const theme = {
+  general: { slogan: 'some-slogan' },
+  logo: 'some-logo',
+  loginPage: { backgroundImg: 'some-image' }
+}
+
+const stubs = {
+  'oc-spinner': true,
+  translate: true
+}
+
+const route = { meta: { title: 'some page title' }, params: { token: 'some-token' } }
+
+const component = { ...PublicLink, mounted: jest.fn(), created: jest.fn() }
+
+const selectors = {
+  loginLogo: '.oc-login-logo',
+  loginCardFooter: '.oc-login-card-footer',
+  loginCardBody: '.oc-login-card-body',
+  submitButton: '.oc-login-authorize-button',
+  errorMessage: '[data-testid="public-link-error-message"]',
+  publicLinkPasswordRequired: '[data-testid="public-link-password-required"]'
+}
+
+const spinnerStub = 'oc-spinner-stub'
+const textInputStub = 'oc-text-input-stub'
+
 describe('PublicLink', () => {
   describe('theming options', () => {
-    test.todo('should have the background image set')
-    test.todo('should display the page title')
-    test.todo('should display the logo image inside login card')
-    test.todo('should display the configuration theme general slogan as the login card footer')
+    const wrapper = getWrapper()
+
+    it('should have the background image set', () => {
+      expect(wrapper).toMatchSnapshot()
+    })
+    it('should display the page title', () => {
+      expect(wrapper).toMatchSnapshot()
+    })
+    it('should display the logo image inside login card', () => {
+      const logo = wrapper.find(selectors.loginLogo)
+
+      expect(logo).toMatchSnapshot()
+    })
+    it('should display the configuration theme general slogan as the login card footer', () => {
+      const slogan = wrapper.find(selectors.loginCardFooter)
+
+      expect(slogan).toMatchSnapshot()
+    })
   })
 
   describe('when the view is still loading', () => {
-    test.todo('should display the loading text with the spinner')
-    test.todo('should not display the error message')
-    test.todo('should not display the password required form')
+    const wrapper = getWrapper({ loading: true })
+    it('should display the loading text with the spinner', () => {
+      const loading = wrapper.find(selectors.loginCardBody)
+
+      expect(wrapper.find(spinnerStub).exists()).toBeTruthy()
+      expect(loading).toMatchSnapshot()
+    })
+    it('should not display the error message', () => {
+      expect(wrapper.find(selectors.errorMessage).exists()).toBeFalsy()
```

Comment:
@pascalwengerter @kulmann for this case is it okay to match the whole wrapper to check that the error message does not exist here?

Reply:
Yes